src/techs/techs.service.ts

```diff
@@ -163,53 +189,245 @@ export class TechsService {
         }
     }
 
-    async addExistingTechVote(req, teamId, teamTechId) {
+    async updateExistingTeamTech(
+        req: CustomRequest,
+        updateTeamTechDto: UpdateTeamTechDto,
+        teamTechItemId: number,
+    ) {
+        const { techName } = updateTeamTechDto;
         // check if team tech item exists
         const teamTechItem = await this.prisma.teamTechStackItem.findUnique({
             where: {
-                id: teamTechId,
+                id: teamTechItemId,
+            },
+            select: {
+                addedBy: {
+                    select: {
+                        member: {
+                            select: {
+                                id: true,
+                            },
+                        },
+                    },
+                },
+                teamTechStackItemVotes: {
+                    select: {
+                        votedBy: {
+                            select: {
+                                member: {
+                                    select: {
+                                        id: true,
+                                    },
+                                },
+                            },
+                        },
+                    },
+                },
             },
         });
+        if (!teamTechItem)
+            throw new NotFoundException(
+                `[Tech Service]: Team Tech Stack Item with id:${teamTechItemId} not found`,
+            );
+
+        // check if the tech stack item has votes other than the user created it
+        if (teamTechItem.teamTechStackItemVotes.length > 1) {
```

Comment:
also here I'm getting an error talking about Property 'teamTechStackItemVotes' does not exist on type.